assets/js/src/tracking.js

```diff
@@ -0,0 +1,219 @@
+import { __ } from '@wordpress/i18n';
+import {
+	getProductFieldObject,
+	getProductImpressionObject,
+	formatPrice,
+} from './utils';
+
+/**
+ * Variable holding the current checkout step. It will be modified by trackCheckoutOption and trackCheckoutStep methods.
+ *
+ * @type {number}
+ */
+let currentStep = -1;
+
+/**
+ * Tracks view_item_list event
+ *
+ * @param {Object} params The function params
+ * @param {Array} params.products The products to track
+ * @param {string} [params.listName] The name of the list in which the item was presented to the user.
+ */
+export const trackListProducts = ( {
+	products,
+	listName = __( 'Product List', 'woocommerce-google-analytics-integration' ),
+} ) => {
+	trackEvent( 'view_item_list', {
+		event_category: 'engagement',
```

Comment:
It looks like all these events are structured for UA whereas GA4 drops the `event_category` and `event_label` structure in favor of individual parameters for each event. Ref: https://developers.google.com/analytics/devguides/collection/ga4/reference/events?client_type=gtag#view_item_list